Source/JavaScriptCore/runtime/JSIteratorPrototype.cpp

```diff
@@ -123,4 +128,39 @@ JSC_DEFINE_CUSTOM_SETTER(iteratorProtoToStringTagSetter, (JSGlobalObject* global
     return JSValue::encode(jsUndefined());
 }
 
+// https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype.toarray
+JSC_DEFINE_HOST_FUNCTION(iteratorProtoFuncToArray, (JSGlobalObject* globalObject, CallFrame* callFrame))
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    JSValue thisValue = callFrame->thisValue();
+    if (!thisValue.isObject())
+        return throwVMTypeError(globalObject, scope, "Iterator.prototype.toArray requires that |this| be an Object."_s);
+
+    MarkedArgumentBuffer value;
+    if (getIterationMode(vm, globalObject, thisValue) == IterationMode::FastArray) {
+        auto* array = jsCast<JSArray*>(thisValue);
+        for (unsigned index = 0; index < array->length(); ++index) {
+            JSValue nextItem = array->getIndex(globalObject, index);
+            RETURN_IF_EXCEPTION(scope, { });
+            value.append(nextItem);
+            if (UNLIKELY(value.hasOverflowed()))
+                throwOutOfMemoryError(globalObject, scope);
+        }
```

Comment:
Ah, I mean, we should have special cloning path for JSArray. In this path, we should not use these getIndex etc. Instead we should just clone butterfly and generate JSArray from that butterfly. We need to care hole existence etc. too, but it is worth doing. You can check some code in FromFast etc.

Reply:
After reading the implementation of `arrayFromFast`, I now understand what you're saying. If possible, could I work on adding that fast path in a separate pull request? I can create the PR within a few days.

I've removed this part for now.